### ISSUE TITLE
warn that profiles are fitted on ML not REML? Allow profiling on REML scale?

### ISSUE THREAD

**Issue by bbolker** [lme4/lme4#325]
As far as I can see, profiling on `lmerMod` objects only works on the ML and not the REML scale.  I don't know if there's theory that says explicitly that profiling the REML criterion does, or does not, produce a meaningful/interpretable object.  However, it does seem at least reasonable that users could be legitimately interested in an interval estimate/confidence interval for REML estimates.  Simply allowing the option to turn off `fm <- refitML(fm)` (and subsequent care with the scale of the objective function), might do it, but  we should think carefully 

Barring this, should we add a warning when computing profiles for REML fits that the model is being refitted with ML?  (I ask because a spent up to an hour myself wondering why the zeros of the profiles were not matching the (REML) point estimates ...)

**Comment by dmbates** [comment]
The justification for the confidence intervals, etc. derived from profiled log-likelihoods is that they are inversions of likelihood-ratio tests.  I have seen some discussion of testing using REML fits in tests but I don't know much about it.  I know that Hong-Suk Ahn did his Ph.D. thesis with Greg Reinsel in our department on that but that work may not have been published.  Greg died suddenly and before Ahn finished.

**Comment by bbolker** [comment]
Thanks, @dmbates.  Do you have any further idea what people do for interval estimates/uncertainties of variance components under REML ... ? (`lme` provides Wald intervals computed on the log-sigma scale, but of course you may not think that's a good idea ...)

**Comment by dmbates** [comment]
@bbolker I'm not really sure what would be done with REML fits.  Many software packages create standard errors for variance components and use them to create confidence intervals and perform hypothesis tests, even though the approach is completely nonsensical.

I'm sure that an argument can be (and probably has been) made for evaluating the difference in the REML criterion and assuming it follows a chi-squared distribution but I don't recall seeing one myself.

As you know, I'm not a big fan of the REML criterion because I don't think that unbiasedness of an estimator of a variance is an important consideration.

**Comment by mmaechler** [comment]
I agree with @dmbates that we shouldn't go into profiling REML estimates before there is good mathematical theory available about that, and hence possibly never.
I agree with @bbolker that it makes sense notifying the user when we ML-refit. I really think giving a `message()` {not a warning} would be more appropriate {and probably even add an option for *not* printing the message.

**Comment by bwiernik** [comment]
@bbolker Have your thoughts on this issue developed at all?

**Comment by bbolker** [comment]
Well, this just came up on the r-sig-mixed-models list (which is probably how you came to it). I had a *brief* look for papers explaining/justifying likelihood profile CIs based on the REML criterion but didn't find any (the ones I found all rely on Wald intervals [which as discussed lots by Doug Bates require very large sample sizes before their assumptions start to be reasonable ...]).

I definitely think a message is a good idea and should implement it shortly (shouldn't be hard ...)

**Comment by bwiernik** [comment]
According to this work by Wolfgang, profiling the REML statistic also follows a chi-square distribution. https://DOI.org/10.1002/sim.2514

So, my recommendation would be to profile the REML statistic rather than refitting with ML

**Comment by bbolker** [comment]
Detailed info from Wolfgang Viechtbauer on `r-sig-mixed-models`:

> After simulating a bunch of datasets, I was able to find one where this issue arises.

```
##############################

dat <- structure(list(id = c(1L, 1L, 1L, 2L, 2L, 2L, 2L, 2L, 3L, 3L, 3L, 3L,
4L, 5L, 5L, 5L, 5L, 5L, 6L, 6L, 6L, 6L, 6L, 7L, 7L, 8L, 9L, 9L, 9L, 9L, 10L,
10L, 10L), yij = c(1.3342, -1.3854, -2.3733, -0.2884, -0.567, -0.6045, 0.494,
0.7738, 0.7991, -0.2539, -0.3956, 1.8508, 2.0849, 0.8782, 1.1726, 0.0818,
0.4099, 0.2279, 0.3244, -0.4167, -0.998, -0.9063, 0.9197, 0.806, -0.1283,
4.1764, -0.0946, -0.2065, 0.0359, 1.2685, -0.0226, 0.6875, 0.1928), obs =
c(1L, 2L, 3L, 1L, 2L, 3L, 4L, 5L, 1L, 2L, 3L, 4L, 1L, 1L, 2L, 3L, 4L, 5L, 1L,
2L, 3L, 4L, 5L, 1L, 2L, 1L, 1L, 2L, 3L, 4L, 1L, 2L, 3L )), row.names = c(NA,
-33L), class = "data.frame")

library(lme4)
library(nlme)
library(metafor)

res1 <- lmer(yij ~ 1 + (1 | id), data=dat)
res2 <- lme(yij ~ 1, random = ~ 1 | id, data=dat)
res3 <- rma.mv(yij ~ 1, V=0, random = ~ 1 | id/obs, data=dat)

summary(res1)
summary(res2)
summary(res3)

# so we get the same results from all fits

# profile likelihood CI for sigma based on ML estimation
confint(res1, parm=".sigma")

# the CI for sigma does not include the REML estimate of sigma
sigma(res1)

# get the profile likelihood CI for sigma based on REML estimation
confint(res3, sigma2=2)

# examine the REML profile for sigma^2
ci <- confint(res3, sigma2=2)
profile(res3, sigma2=2, cline=TRUE, steps=100, xlim=c(0.4,2.2))
abline(v=ci$random[1,2:3], lty="dotted")

# note: for this small dataset, the ML fit is really rather different
summary(update(res1, REML=FALSE))

# double-check the profile likelihood CI based on ML estimation
res3 <- update(res3, method="ML")
res3
confint(res3, sigma2=2)

# the lower bound is rather different compared to confint(res1, parm=".sigma")

# examine the ML profile for sigma^2
ci <- confint(res3, sigma2=2)
profile(res3, sigma2=2, cline=TRUE, steps=100, xlim=c(0.4,2.2))
abline(v=ci$random[1,2:3], lty="dotted")

##############################
```

> So, as far as I can tell, there are really two issues here:
> 1) As noted already, confint.merMod() uses the ML likelihood to profile all parameters including the variance components. This can in principle lead to the CI not including a particular REML estimate of a parameter. However, the switch to ML makes sense in so far as confint.merMod() also profiles fixed effects for which one must use ML estimation. It would be *really* confusing if the CIs for the fixed effects are based on the regular while the CIs for variance components are based on restricted likelihood. Might be nice though to emphasize under help(confint.merMod) that this is happening.

> 2) However, in the example above, the profiling itself goes wrong, leading to an incorrect lower bound of the profile likelihood CI for sigma. After playing around with some of the settings to profile.merMod(), I got the same CI as metafor with:

```r
confint(res1, parm=".sigma", delta.cutoff=1/64)
```

>So, @Emmanuel, you might want to check if the issue you are encountering is not one of type 1) but rather of type 2).

**Comment by bwiernik** [comment]
Interesting. Personally, I don't think it is that confusing to use restricted likelihood profiling for the variance components and likelihood profiling for the fixed effects. That makes sense to me and is rather consistent with sometimes-recommended LRT procedures to compared models differing only in random effects using restricted likelihood, but likelihood otherwise
